File: API/Helpers/UserParams.cs
namespace API.Helpers
{
    public class UserParams : PaginationParams
    {
        public string CurrentUsername { get; set; }
        public string Gender { get; set; }
        // public int MinAge { get; set; } = 18;
        // public int MaxAge { get; set; } = 150;
        public string OrderBy { get; set; } = "lastActive";
        public string Country { get; set; }
        public string City { get; set; }
        public string Major { get; set; }
        public bool isVerified { get; set; }
        // for admin search
        public string Username { get; set; }

        
    }
}

File: client/src/app/_models/userParams.ts
import { User } from './user';

export class UserParams {
    gender: string;
    // minAge = 18;
    // maxAge = 99;
    pageNumber = 1;
    pageSize =24;
    orderBy = 'lastActive';

    country: string;
    city: string;
    major: string;
    // isVerified: boolean;
    constructor(user: User) {
 
        // this.gender = user.gender === 'female' ? 'male' : 'female';
        
        // initial values to be passed in URL to show all since it exclude condition in UserRepository GetMembers filter
        this.gender =  'الكل';
        this.country = 'جميع البلدان'
        this.city = 'جميع المدن'
        this.major = 'جميع التخصصات'
        // this.pageSize =5;
    }
}